logs-to-tsv/query-logs.js

```diff
@@ -0,0 +1,83 @@
+const { CloudWatchLogsClient, StartQueryCommand, GetQueryResultsCommand } = require("@aws-sdk/client-cloudwatch-logs")
+const { writeFileSync } = require('fs')
+const aws = require('aws-sdk')
+
+const awsInit = (profile) => {
+  // Set aws creds:
+  aws.config.credentials = new aws.SharedIniFileCredentials({
+    profile: profile || 'nypl-digital-dev'
+  })
+
+  // Set aws region:
+  const awsSecurity = { region: 'us-east-1' }
+  aws.config.update(awsSecurity)
+}
+
+awsInit()
+
+const client = new CloudWatchLogsClient();
+
+const start = new Date();
+start.setMonth(start.getMonth() - 3)
+
+const inputs = []
+for (let i = 0; i < 20; i++) {
+  inputs.push({
+    "startTime": Date.parse(start),
+    "endTime": start.setDate(start.getDate() + 2),
+    "logGroupName": "/aws/elasticbeanstalk/discovery-api-production/var/log/nginx/access.log",
+    "queryString": "fields @message | sort @timestamp desc | filter @message like \"/resources?\" | filter @message not like \"filters\" | limit 10000 | filter @message not like \"oclc\"",
+  })
+}
+
+const delay = async (time) => new Promise((resolve) => setTimeout(resolve, time))
+
+const checkResultStatusAndDownload = async (queryIds, completedQueries = {}, numberOfChecks = 1) => {
+  while (queryIds.length !== Object.keys(completedQueries).length) {
+    console.log('starting check ' + numberOfChecks)
+    await Promise.all(queryIds.map(async (queryId, i) => {
+      // Ensure there are less than 5 requests per second to avoid throttling
+      await delay(220)
```

Comment:
I don't think this `delay` is doing anything as the function is written. You're looping over `queryIds` and firing off async calls and each of those calls is waiting for 220ms and then proceeding at roughly the same time; A delay in the first call does not postpone the second call, etc. I think it can be removed. If AWS has a calls/s threshold, I've not hit it. It does appear to enforce an active-queries threshold, but I don't think the above `delay` would help that.